xrpl/model/transactions/validations_xrpl_objects.go

```diff
@@ -87,12 +78,47 @@ func IsAmount(amount interface{}) bool {
 	return false
 }
 
+// IsAsset checks if the given object is a valid Asset object. Typically used in AMM.
+// https://xrpl.org/docs/references/protocol/ledger-data/ledger-entry-types/amm#amm-fields
+func IsAsset(input interface{}) bool {
+	if !typecheck.IsMap(input) {
```

Comment:
We can remove this type check since it is checked on line 88, don't forget to add the `ok` field